### ISSUE TITLE
Fresh AngualrDart project fails to build.

### ISSUE THREAD

**Issue by kdxvc** [dart-lang/stagehand#522]
[**Edit** by @chalin - trimmed down the log since we don't need the whole thing to diagnose the problem]

When starting a new AngularDart project via stagehand project template, new project fails to build with errors.

C:\Users\its_me>dart --version
Dart VM version: 2.0.0-dev.58.0 (Wed May 23 20:44:51 2018 +0200) on "windows_x64"

C:\Users\its_me>pub --version
Pub 2.0.0-dev.58.0

intellij dart plugin v182.2371.1

How to reproduce.

1) open phpstorm
2) File > New Project...
3) select Dart > Generate sample content > AngularDart Web App > Create
4) new project opens in IDE
5) run pub get
```
>pub get
Resolving dependencies...
Overriding the upper bound Dart SDK constraint to <=2.0.0-dev.58.0 for the following packages:

archive, args, bazel_worker, boolean_selector, build_config, charcode, cli_util, convert, fixnum, glob, graphs, html, http, http_multi_server, http_parser, intl, io, js, logging, meta, multi_server_socket, node_preamble, package_config, path, perf_api, plugin,
protobuf, pub_semver, quiver, quiver_hashcode, sass, shelf_packages_handler, source_map_stack_trace, source_maps, source_span, stack_trace, stream_channel, string_scanner, term_glyph, tuple, typed_data

To disable this you can set the PUB_ALLOW_PRERELEASE_SDK system environment variable to `false`, or you can silence this message by setting it to `quiet`.
+ analyzer 0.32.0
+ angular 5.0.0-alpha+12 (5.0.0-alpha+13 available)
+ angular_ast 0.5.3+1
+ angular_components 0.9.0-alpha+12 (0.9.0-alpha+13 available)
+ angular_forms 2.0.0-alpha+4 (2.0.0-alpha+5 available)
+ angular_test 2.0.0-alpha+10 (2.0.0-alpha+11 available)
+ archive 1.0.33 (2.0.0 available)
+ args 1.4.3
+ async 2.0.7
+ bazel_worker 0.1.9 (0.1.10 available)
+ boolean_selector 1.0.3
+ build 0.12.6
+ build_config 0.2.6+2
+ build_modules 0.2.2+5
+ build_resolvers 0.2.0+2
+ build_runner 0.8.9
+ build_test 0.10.2+3
+ build_web_compilers 0.4.0+1
+ built_collection 3.1.1
+ built_value 5.5.0
+ charcode 1.1.1
+ cli_util 0.1.3
+ code_builder 3.0.3
+ collection 1.14.9
+ convert 2.0.1
+ crypto 2.0.3
+ csslib 0.14.4
+ dart2_constant 1.0.1+dart2
+ dart_internal 0.1.1
+ dart_style 1.0.14
+ fixnum 0.10.7
+ front_end 0.1.0
+ glob 1.1.5
+ graphs 0.1.2
+ html 0.13.3
+ http 0.11.3+16
+ http_multi_server 2.0.4
+ http_parser 3.1.2
+ intl 0.15.6
+ io 0.3.2+1
+ js 0.6.1
+ json_annotation 0.2.5
+ kernel 0.3.0
+ logging 0.11.3+1
+ matcher 0.12.2+1
+ meta 1.1.5
+ mime 0.9.6+1
+ multi_server_socket 1.0.1
+ node_preamble 1.4.2
+ observable 0.22.1+3
+ package_config 1.0.3
+ package_resolver 1.0.3
+ path 1.5.1
+ perf_api 0.1.0
+ plugin 0.2.0+2
+ pool 1.3.5
+ protobuf 0.7.2 (0.9.0 available)
+ pub_semver 1.4.1
+ quiver 0.29.0+1
+ quiver_hashcode 1.0.0
+ sass 1.4.0
+ sass_builder 2.0.0
+ scratch_space 0.0.2+1
+ shelf 0.7.3+1
+ shelf_packages_handler 1.0.3
+ shelf_static 0.2.7+1
+ shelf_web_socket 0.2.2+1
+ source_gen 0.8.1+3
+ source_map_stack_trace 1.1.4
+ source_maps 0.10.5
+ source_span 1.4.0
+ stack_trace 1.9.2
+ stream_channel 1.6.6
+ stream_transform 0.0.11
+ string_scanner 1.0.2
+ term_glyph 1.0.0
+ test 0.12.38+1
+ tuple 1.0.1
+ typed_data 1.1.5
+ utf 0.9.0+4
+ watcher 0.9.7+8
+ web_socket_channel 1.0.7
+ yaml 2.1.14
Warning: You are using these overridden dependencies:
! angular_compiler 0.4.0-alpha+12 (0.4.0-alpha+13 available)
Changed 84 dependencies!
Precompiling executables...
Precompiled build_runner:graph_inspector.
Precompiled build_runner:build_runner.
Precompiled test:test.

```
6) run index.html  which kicks off Dart Dev Server

7) watch everything fail while trying to compile/serve the new project
```
C:\Programming\dart\sdk\bin\pub.bat global run webdev serve web:54505
webdev could not run for this project.
No pubspec.lock file found, please run "pub get" first.
Dart Dev Server terminated
C:\Programming\dart\sdk\bin\pub.bat global run webdev serve web:54535
[INFO] Setting up file watchers...
[INFO] Setting up file watchers completed, took 19ms

[INFO] Waiting for all file watchers to be ready...
[INFO] Waiting for all file watchers to be ready completed, took 12ms

[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 1.6s

[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 1ms

[INFO] Running build...
[SEVERE] angular_components|scss_builder on package:angular_components/material_checkbox/material_checkbox.scss:

Error: Can't find stylesheet to import.
@import 'const/global',
        ^^^^^^^^^^^^^^
  c:\users\its_me\appdata\local\temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\css\material\_material.scss 19:9              @import
  C:\Users\its_me\AppData\Local\Temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\material_checkbox\material_checkbox.scss 5:9  root stylesheet
[SEVERE] angular_components|scss_builder on package:angular_components/material_input/material_auto_suggest_input.scss:

Error: Can't find stylesheet to import.
@import 'const/global',
        ^^^^^^^^^^^^^^
  c:\users\its_me\appdata\local\temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\css\material\_material.scss 19:9                     @import
  C:\Users\its_me\AppData\Local\Temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\material_input\material_auto_suggest_input.scss 5:9  root stylesheet
[SEVERE] angular_components|scss_builder on package:angular_components/material_input/material_input.scss:

Error: Can't find stylesheet to import.
@import 'const/global',
        ^^^^^^^^^^^^^^
  c:\users\its_me\appdata\local\temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\css\material\_material.scss 19:9        @import
  C:\Users\its_me\AppData\Local\Temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\material_input\material_input.scss 5:9  root stylesheet
[SEVERE] angular_components|scss_builder on package:angular_components/material_chips/material_chips.scss:

Error: Can't find stylesheet to import.
@import 'const/global',
        ^^^^^^^^^^^^^^
  c:\users\its_me\appdata\local\temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\css\material\_material.scss 19:9        @import
  C:\Users\its_me\AppData\Local\Temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\material_chips\material_chips.scss 5:9  root stylesheet
[SEVERE] angular_components|scss_builder on package:angular_components/material_datepicker/material_date_time_picker.scss:

Error: Can't find stylesheet to import.
@import 'const/global',
        ^^^^^^^^^^^^^^
  c:\users\its_me\appdata\local\temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\css\material\_material.scss 19:9                        @import
  C:\Users\its_me\AppData\Local\Temp\scratch_spacea99afdde-6366-11e8-a683-005056c00008\packages\angular_components\material_datepicker\material_date_time_picker.scss 5:9  root stylesheet
[SEVERE] angular_components|scss_builder on package:angular_components/highlighted_text/highlighted_text.scss:

[...]

Please fix all errors before compiling (warnings are okay).
}
[WARNING] build_web_compilers|entrypoint on web/main.dart:
Unable to read angular_components|lib/scorecard/scoreboard.template.ddc.js, check your console or the `.dart_tool/build/generated/angular_components/lib/scorecard/scoreboard.template.ddc.js.errors` log file.
[INFO] 1m 29s elapsed, 2757/2773 actions completed.
[INFO] 1m 30s elapsed, 2760/2773 actions completed.
[INFO] 1m 31s elapsed, 2764/2773 actions completed.
[INFO] 1m 32s elapsed, 2765/2773 actions completed.
[INFO] 1m 33s elapsed, 2773/2778 actions completed.
[INFO] 1m 34s elapsed, 7353/7353 actions completed.
[INFO] Running build completed, took 1m 34s

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 342ms

Serving `web` on http://localhost:54535
[SEVERE] Failed after 1m 37s
```

**Comment by bwnyasse** [comment]
Hi @chalin  I've just test it and it's working for me.  👍 

When overriding sass version , the fresh Angular Dart generated by stagehand is working

```
dependency_overrides:
  sass: 1.3.2
```

Following the ouput of `webdev build`:

```
[INFO] Reading cached asset graph completed, took 1.3s
[INFO] Checking for updates since last build completed, took 1.2s
[WARNING] Invalidating asset graph due to build script update
[INFO] Cleaning up outputs from previous builds. completed, took 136ms
[INFO] Building new asset graph completed, took 2.2s
[INFO] Checking for unexpected pre-existing outputs. completed, took 2ms
[INFO] build_web_compilers|entrypoint on web/main.dart:Running dart2js with --minify --packages=.package-55f00cc37793a05d37531ccf0d980c39 -oweb/main.dart.js web/main.dart
[INFO] build_web_compilers|entrypoint on web/main.dart:Running dart2js with --minify --packages=.package-851618ffcc18441b4904d97f20be3bb8 -otest/app_test.dart.js test/app_test.dart
[INFO] build_web_compilers|entrypoint on web/main.dart:Dart2Js finished with:

Compiled 12,930,501 characters Dart to 276,948 characters JavaScript in 14.1 seconds
Dart file (web/main.dart) compiled to JavaScript: web/main.dart.js

[WARNING] No actions completed for 15.0s, waiting on:
  - build_web_compilers|entrypoint on test/app_test.dart
  - build_web_compilers|entrypoint on test/app_test.dart.browser_test.dart

[INFO] build_web_compilers|entrypoint on web/main.dart:Running dart2js with --minify --packages=.package-3b0023b8b0bfdb347cc595cba455e130 -otest/app_test.dart.browser_test.dart.js test/app_test.dart.browser_test.dart
[INFO] build_web_compilers|entrypoint on test/app_test.dart:Dart2Js finished with:

Compiled 13,773,714 characters Dart to 505,805 characters JavaScript in 15.9 seconds
Dart file (test/app_test.dart) compiled to JavaScript: test/app_test.dart.js

[WARNING] No actions completed for 15.1s, waiting on:
  - build_web_compilers|entrypoint on test/app_test.dart.browser_test.dart

[INFO] build_web_compilers|entrypoint on test/app_test.dart.browser_test.dart:Dart2Js finished with:

Compiled 13,791,392 characters Dart to 566,127 characters JavaScript in 19.6 seconds
Dart file (test/app_test.dart.browser_test.dart) compiled to JavaScript: test/app_test.dart.browser_test.dart.js

[INFO] Running build completed, took 1m 37s
[INFO] Caching finalized dependency graph completed, took 1.3s
[INFO] Succeeded after 1m 39s with 1646 outputs (5673 actions)
```

**Comment by chalin** [comment]
> How do I know if/when I will be able to remove the version overrides?

@kdxvc - watch / subscribe to https://github.com/dart-lang/stagehand/issues/524

**Comment by chalin** [comment]
Sass 1.5.0 has been released with a fix, and so #524 has been closed.
Leaving this open until a new version of stagehand is published. /cc @kevmoo

**Comment by chalin** [comment]
This should be resolved in the latest release.
